desk/app/groups.hoon

```diff
@@ -1435,49 +1429,63 @@
     go-core
   ::
   ++  go-watch
-    |=  [ver=?(%v0 %v1) =(pole knot)]
+    |=  [ver=?(%v0 %v1 %v2) =(pole knot)]
     ^+  go-core
     ?+    pole  !!
-        [%updates rest=*]  (go-pub ver rest.pole)
-        [%ui ~]            go-core
-        [%preview ~]       (go-preview ver)
+      ::
+        [%updates rest=*]
+      ?>  ?=(?(%v0 %v1) ver)
+      (go-pub ver rest.pole)
     ::
-        ::XX seems defunct, remove
-        [%bait host=@ name=@ ~]
-      ?>  ?=(%open -.cordon.group)
-      =/  =flag:g  [(slav %p host.pole) name.pole]
-      =;  =nest:g
-        =.  cor  (give %fact ~ dude+!>(p.nest))
-        =.  cor  (give %kick ~ ~)
-        go-core
-      %-  need
-      %+  roll  ~(tap in imported.group)
-      |=  [=nest:g out=(unit nest:g)]
-      ^-  (unit nest:g)
-      ?.  =(~ out)  out
-      ?.  =(q.nest flag)  ~
-      `nest
+        [%ui ~]       ?>(?=(?(%v0 %v1) ver) go-core)
+        [%preview ~]  (go-preview ver)
     ==
   ::
   ++  go-preview
-    |=  ver=?(%v0 %v1)
-    ?>  ?-  -.cordon.group
+    |=  ver=?(%v0 %v1 %v2)
+    =/  allow=?
+      ?-  -.cordon.group
           %afar  &
           %open  !secret.group  :: should never be secret
         ::
             %shut
-          ::  if a private group yes
-          ::  if secret, only invites should get previews
-          ?.  secret.group  &
-          (~(has in pend.cordon.group) src.bowl)
+          ::  allow previews of a private group:
+          ::  (1) if it is *not* secret, or
+          ::  (2) the viewer is on the invitation list
+          ::
+          ?|  !secret.group
+              (~(has in pend.cordon.group) src.bowl)
+          ==
         ==
+    ::  access control: crash if we are on v0, v1
+    ::  and we disallow the preview
+    ::
+    ?<  &(?=(%v0 %v1) !allow)
     =/  =preview:g
       =,  group
       [flag meta cordon now.bowl secret.group ~(wyt by fleet)]
     =.  cor
-      ?:  ?=(%v0 ver)
+      ?-    ver
+          %v0
         (emit %give %fact ~ group-preview+!>((to-preview-2 preview)))
-      (emit %give %fact ~ group-preview-1+!>(preview))
+      ::
+          %v1
+        =/  =preview:v5:g  preview
+        (emit %give %fact ~ group-preview-1+!>(preview))
+      ::
+          %v2
+        ?.  allow
+          ?:  secret.group
+            ::  conceal secret private group
+            ::
+            =/  pev  `preview-response:v6:g`|+%missing
```

Comment:
If you're gonna `=/` might as well make use of it. (;

```suggestion
            =/  pev=preview-response:v6:g  |+%missing
```

Reply:
oh, definitely!